openhtf/__init__.py

```diff
@@ -71,21 +72,39 @@ def PhaseTwo(test):
       # Analyze widget integration status
 
     Test(PhaseOne, PhaseTwo).Execute()
+
+  Note that Test() objects *must* be created in the main thread, but can be
+  .Execute()'d in a separate thread.
   """
 
   def __init__(self, *phases, **metadata):
     code_info = test_record.CodeInfo.ForModuleFromStack(levels_up=2)
+    self._test_options = TestOptions()
     self._test_info = TestData(phases, metadata=metadata, code_info=code_info)
-    self._output_callbacks = []
-    self.loop = False
+    self._lock = threading.Lock()
+    self._stopped = False
+    # Make sure Configure() gets called at least once before Execute().  The
+    # user might call Configure() again to override options, but we don't want
+    # to force them to if they want to use defaults.  For default values, see
+    # the class definition of TestOptions.
+    self.Configure()
+    # TODO(madsci): Fix this to play nice with multiple Test instances.
+    signal.signal(signal.SIGINT, self.Stop)
 
   def AddOutputCallback(self, callback):
+    """DEPRECATED: Use AddOutputCallbacks() instead."""
+    # TODO(madsci): Remove this before we push to PyPI, here for transitionary
+    # purposes. 
+    raise AttributeError(
+        'DEPRECATED, use AddOutputCallbacks() instead of AddOutputCallback()')
+
+  def AddOutputCallbacks(self, *callbacks):
     """Add the given function as an output module to this test."""
-    self._output_callbacks.append(callback)
+    self._test_options.output_callbacks.extend(callbacks)
```

Comment:
I went back and forth in my head on this one for a while, and I'm okay with keeping both ways in the mix. The value in the ease of having `AddOutputCallbacks` it worth it in my opinion because almost everyone needs output callbacks on almost every test.